source/device/tensorrt/trt_executor.cc

```
@@ -526,45 +588,51 @@ int TensorRTEngine::PreRun(struct subgraph* subgraph, struct trt_option* options
         TLOG_ERR("Tengine: Cannot using GPU ID %d.\n", opt->gpu_index);
         return -2;
     }
-
     struct graph* ir_graph = subgraph->graph;
-
     // set tensor_swap_count
     this->tensor_swap_count = subgraph->graph->tensor_num + 1;
-
-    // TODO: high level api should serialize model and cache serialized model
-    this->builder = nvinfer1::createInferBuilder(gLogger.get_logger());
-    if (nullptr == this->builder)
+    if (DeserializeEngine(opt->engine_file, &this->engine))
     {
-        TLOG_ERR("Tengine: Cannot create nvinfer1::IBuilder object..\n");
-        return -3;
+        TLOG_ERR("Tengine: DeserializeEngine success.\n");
     }
-
-    // TODO: adapt to TRT6
-    const auto _explicit_batch = 1U << static_cast<uint32_t>(nvinfer1::NetworkDefinitionCreationFlag::kEXPLICIT_BATCH);
-    this->network = this->builder->createNetworkV2(_explicit_batch);
-    if (nullptr == this->network)
+    else
     {
-        TLOG_ERR("Tengine: Cannot create nvinfer1::INetworkDefinition object.\n");
-        return -4;
-    }
+        TLOG_ERR("Tengine: DeserializeEngine failed,build engine.\n");
```

should judge pointer firstly, if is nullprt, no need to report an error.